components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Discovery/src/Services/DiscoveryServices.cs

```diff
@@ -491,13 +492,18 @@ public class DiscoveryServices : IDiscoveryServices, IScannerServices, IDisposab
                         // Check local host
                         if (host.EqualsIgnoreCase("localhost") && Host.IsContainer) {
                             // Also resolve docker internal since we are in a container
-                            host = kDockerHostName;
+                            host = Environment.GetEnvironmentVariable(IoTEdgeVariables.IOTEDGE_GATEWAYHOSTNAME);
                             continue;
                         }
                         break;
                     }
-                    catch (Exception e) {
-                        _logger.Warning(e, "Failed to resolve the host for {discoveryUrl}", discoveryUrl);
+                    catch (SocketException se) {
+                        _logger.Warning("Failed to resolve the host for {discoveryUrl} due to {message}",
+                            discoveryUrl, se.Message);
+                        return list;
```

Comment:
shouldn't this catch be in the loop and we should continue to resolve other adresses?

Reply:
I find this construction rather weird... Actually we only have a single "host" string as argument, so the while is superflous. We want though to handle the special case when this argument is "localhost". "Localhost" in the module context makes no sense but we want to target the discovery call to the iotedge host system itself. I would immediately get rid of the "while" and handle "localhost" redirection before resolving the host. But, I am not a fan of on-the-fly off-context refactorings so I'll leave it be,  for now :)

Reply:
thanks for the clarification